### PR TITLE
Added `ProductVariantModel.imageVariants` getter

### LINKED ISSUE #10: Image sizes
Images returned from the API are the raw image size uploaded to Shopify, i.e. 
`https://cdn.shopify.com/s/files/1/1015/1595/products/blackhelmet_store.png?v=1446233645`

We should be able to return these in the various sizes Shopify produces, which requires adding a suffix to the file name matching [these sizes](https://docs.shopify.com/themes/liquid/filters/url-filters#size-parameters). 

i.e. instead of `product.selectedVariantImage.src`, maybe `product.selectedVariantImage('thumb')`?

@minasmart @tessalt @harismahmood89

### PR DESCRIPTION
**Pull request by yomexzo**
Fixes #10 and it's part of the v0.2.0 list at #51.

Format is gotten from Shopify/shopify#71114

@minasmart, @harismahmood89, @tessalt

### CODE REVIEW

File: src/models/product-variant-model.js [hunk 1]
```diff
@@ -103,6 +103,37 @@ const ProductVariantModel = BaseModel.extend({
...
+      { name: 'medium', dimension: '240x240' },
+      { name: 'large', dimension: '480x480' },
+      { name: 'grande', dimension: '600x600' },
+      { name: '1024x1024', dimension: '1024x1024' },
+      { name: '2048x2048', dimension: '2048x2048' }
+    ];
+
+    variants.forEach((variant, index) => {
+      variant.src = `${srcParts[0]}_${variant.name}${srcParts[1]}`;
+      variants[index] = variant;
```
Comment (minasmart) [inline src/models/product-variant-model.js]: This line doesn't need to exist.

File: src/models/product-variant-model.js [hunk 2]
```diff
@@ -103,6 +103,37 @@ const ProductVariantModel = BaseModel.extend({
...
   /**
+    * Image variants available for a variant, ex [ {"name":"pico","dimension":"16x16","src":"https://cdn.shopify.com/image-two_pico.jpg"} ]
+    * See https://help.shopify.com/themes/liquid/filters/url-filters#size-parameters for list of available variants.
+    * @property imageVariant
+    * @type {Array}
+  */
+  get imageVariants() {
+    const src = this.image.src;
+    const extensionIndex = src.lastIndexOf('.');
+    const srcParts = [src.slice(0, extensionIndex), src.slice(extensionIndex)];
```
Comment (minasmart) [inline src/models/product-variant-model.js]: Two named `const`s please 😄

File: tests/unit/models/product-variant-test.js [hunk 3]
```diff
@@ -49,6 +49,59 @@ const baseAttrs = {
...
+    name: '1024x1024',
+    dimension: '1024x1024',
+    src: 'https://cdn.shopify.com/image-two_1024x1024.jpg'
+  },
+  {
+    name: '2048x2048',
+    dimension: '2048x2048',
+    src: 'https://cdn.shopify.com/image-two_2048x2048.jpg'
+  }
+];
```
Comment (minasmart) [inline tests/unit/models/product-variant-test.js]: I'd just scope this to the test in which it's required.

File: src/models/product-variant-model.js (hunk 1 above)
Comment (yomexzo) [inline src/models/product-variant-model.js]: True, i initially had `variants[index].src`. I felt it's neater. Removing now.

**Comment by tessalt** [comment]
LGTM other than mina's comments

File: src/models/product-variant-model.js (hunk 2 above)
Comment (yomexzo) [inline src/models/product-variant-model.js]: :) .. i had a huge dilemma with the name to give them.

File: src/models/product-variant-model.js (hunk 1 above)
Comment (minasmart) [inline src/models/product-variant-model.js]: You don't even need the index though. `variant` is the element from the array.

File: src/models/product-variant-model.js (hunk 2 above)
Comment (minasmart) [inline src/models/product-variant-model.js]: `extension` and `pathAndBasename`?

**Comment by minasmart** [comment]
👍 

interactive rebase into one commit, and :shipit:

**Comment by tessalt** [comment]
should we enable "merge and squash" button on this repo?

**Comment by minasmart** [comment]
@tessalt will do.

**Pull request merged**